minos/api_gateway/discovery/handlers.py

```diff
@@ -37,49 +35,55 @@ async def get_formatted(request: web.Request) -> dict:
     }
 
 
-class DiscoveryHandlers(object):
-    async def discover(self, request: web.Request, config: MinosConfig, **kwargs):
-        name = request.query.get("name")
-        if not name:
-            return web.json_response('Parameter "name" not found.', status=400)
-        else:
-            # Search by key in Redis and return JSON
-            redis_cli = MinosRedisClient(config=config)
+@routes.get("/subscriptions")
+async def discover(request: web.Request):
+    name = request.query.get("name")
+    if not name:
+        return web.json_response('Parameter "name" not found.', status=400)
+
+    # Search by key in Redis and return JSON
+    redis_cli = MinosRedisClient(config=request.app["config"])
+
+    # Get JSON data
+    data = redis_cli.get_data(name)
+    return web.json_response(data=data)
+
+
+@routes.post("/subscriptions")
+async def subscribe(request: web.Request):
+    validation, errors = await validate_input(request)
 
-            # Get JSON data
-            data = redis_cli.get_data(name)
-            return web.json_response(data=data)
+    if errors:
+        return validation
 
-    async def subscribe(self, request: web.Request, config: MinosConfig, **kwargs):
-        validation, errors = await validate_input(request)
+    input_json = await get_formatted(request)
 
-        if errors:
-            return validation
+    redis_client = MinosRedisClient(config=request.app["config"])
 
-        input_json = await get_formatted(request)
+    redis_client.set_data(input_json["name"], input_json)
```

Comment:
Why not to set the `MinosRedisClient` as a `web.Applicantion` configuration entry, instead of recreating it each time a request comes?

Reply:
This question will be resolved by https://github.com/Clariteia/discovery/pull/46/files#r684969388